### PR TITLE
Refactor cnv

### PR DESCRIPTION
**Pull request by jonca79**

### CODE REVIEW

File: workflow/rules/cnvkit.smk [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+        "{rule}: Plot cnvs into cnv_sv/cnvkit_scatter/{wildcards.sample}_{wildcards.type}.png"
+    shell:
+        "(cnvkit.py scatter {input.segment_regions} "
+        "-s {input.segments} "
+        "-v {input.vcf} "
+        "-o {output.plot} "
+        "{params.extra}) &> {log}"
+
+
+rule cnvkit_vcf:
```
Comment (marrip) [inline workflow/rules/cnvkit.smk:163]: I think this rule is not part of cnvkit so it deserves its own `.smk`

File: workflow/rules/gatk_cnv.smk [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+        mem_per_cpu=config.get("gatk_cnv_denoise_read_counts", {}).get("mem_per_cpu", config["default_resources"]["mem_per_cpu"]),
+        partition=config.get("gatk_cnv_denoise_read_counts", {}).get("partition", config["default_resources"]["partition"]),
+    container:
+        config.get("gatk_cnv_denoise_read_counts", {}).get("container", config["default_container"])
+    conda:
+        "../envs/gatk_cnv.yaml"
+    message:
+        "{rule}: Use gatk_cnv to obtain cnv_sv/gatk_cnv_denoise_read_counts/{wildcards.sample}_{wildcards.type}.clean.denoisedCR.tsv"
+    shell:
+        "(gatk --java-options '-Xmx4g' DenoiseReadCounts -I {input.hdf5Tumor} "
```
Comment (marrip) [inline workflow/rules/gatk_cnv.smk:120]: I think `-I` should get a new line as it is a flag

File: workflow/rules/gatk_cnv.smk [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+    message:
+        "{rule}: Use gatk_cnv to obtain cnv_sv/gatk_cnv_call_copy_ratio_segments/{wildcards.sample}_{wildcards.type}.clean.calledCNVs.seg"
+    shell:
+        "(gatk --java-options '-Xmx4g' CallCopyRatioSegments "
+        "--input {input} "
+        "--output {output.segments} "
+        "{params.extra}) &> {log}"
+
+
+rule gatk_cnv_vcf:
```
Comment (marrip) [inline workflow/rules/gatk_cnv.smk:212]: Here as well, should be an individual `.smk` file

File: workflow/rules/cnvkit.smk (hunk 1 above)
Comment (jonca79) [reply workflow/rules/cnvkit.smk]: done

File: workflow/rules/gatk_cnv.smk (hunk 2 above)
Comment (jonca79) [reply workflow/rules/gatk_cnv.smk]: done

File: workflow/rules/gatk_cnv.smk (hunk 3 above)
Comment (jonca79) [reply workflow/rules/gatk_cnv.smk]: done

**Review by marrip — APPROVED** [review]

**Pull request merged**